File: workspace/2023-07-17/numerical-sequence.forth/fibonacci.fs

: _fibonacci recursive ( a b n -- fn )
    dup 0 = if
        drop drop
    else
        rot rot tuck + rot
        1 - _fibonacci
    endif
;

: fibonacci ( n -- fn )
    0 1 rot _fibonacci
;

: _show_fibonacci_numbers recursive ( max min -- )
    over over < if
        drop drop
    else
        dup fibonacci .
        1 + _show_fibonacci_numbers
    endif
;

: show_fibonacci_numbers ( n -- )
    s\" fibonacci numbers:\n" type
    0 _show_fibonacci_numbers
    s\" \n" type
;

File: workspace/2023-08-22/numerical-sequence.prolog/fibonacci.pl

fibonacci(0, 0).

fibonacci(1, 1).

fibonacci(N, Fn) :-
    is(A, N - 2),
    is(B, N - 1),
    fibonacci(A, Fa),
    fibonacci(B, Fb),
    is(Fn, Fa + Fb).

